app/src/main/java/com/nextcloud/talk/chat/ChatActivity.kt

```diff
@@ -835,6 +871,84 @@ class ChatActivity :
             resources?.getString(R.string.nc_description_send_message_button)
     }
 
+    private fun editMessageAPI(message: ChatMessage, editedMessageText: String) {
+        var apiVersion = 1
+        // FIXME Fix API checking with guests?
+        if (conversationUser != null) {
+            apiVersion = ApiUtils.getChatApiVersion(conversationUser, intArrayOf(1))
+        }
+
+        ncApi.editChatMessage(
+            credentials,
+            ApiUtils.getUrlForChatMessage(
+                apiVersion,
+                conversationUser?.baseUrl,
+                roomToken,
+                message?.id
+            ),
+            editedMessageText
+        )?.subscribeOn(Schedulers.io())
+            ?.observeOn(AndroidSchedulers.mainThread())
+            ?.subscribe(object : Observer<ChatOverallSingleMessage> {
+                override fun onSubscribe(d: Disposable) {
+                    // unused atm
+                }
+
+                override fun onNext(messageEdited: ChatOverallSingleMessage) {
+                    when (messageEdited.ocs?.meta?.statusCode) {
+                        HTTP_BAD_REQUEST -> {
+                            Snackbar.make(
+                                binding.root,
+                                getString(R.string.edit_error_24_hours_old_message),
+                                Snackbar.LENGTH_LONG
+                            ).show()
+                        }
+                        HTTP_FORBIDDEN -> {
+                            Snackbar.make(
+                                binding.root,
+                                getString(R.string.conversation_is_read_only),
+                                Snackbar.LENGTH_LONG
+                            ).show()
+                        }
+                        HTTP_NOT_FOUND -> {
+                            Snackbar.make(
+                                binding.root,
+                                "Conversation not found",
+                                Snackbar.LENGTH_LONG
+                            ).show()
+                        }
+                    }
+                    message.message = messageEdited.ocs?.data?.parentMessage?.text
+                    message.lastEditTimestamp = messageEdited.ocs?.data?.lastEditTimestamp!!
+                    adapter?.update(message)
+                    adapter?.notifyDataSetChanged()
+                    clearEditUI()
+                }
+
+                override fun onError(e: Throwable) {
```

Comment:
Log.e(TAG, "failed to edit message", e)
                Snackbar.make(binding.root, R.string.nc_common_error_sorry, Snackbar.LENGTH_LONG).show()